pkg/adapter/awscodecommitsource/adapter.go

```diff
@@ -216,68 +216,49 @@ func (a *adapter) preparePullRequests() ([]*codecommit.PullRequest, error) {
 		RepositoryName: &a.arn.Resource,
 	}
 
-	for {
-		//Get pull request IDs
-		pullRequestsOutput, err := a.ccClient.ListPullRequests(&input)
-		if err != nil {
-			return pullRequests, fmt.Errorf("failed to list PRs: %w", err)
-		}
-
-		prIDs := []*string{}
-
-		prIDs = append(prIDs, pullRequestsOutput.PullRequestIds...)
-
-		for _, id := range prIDs {
-			pri := codecommit.GetPullRequestInput{PullRequestId: id}
-
-			prInfo, err := a.ccClient.GetPullRequest(&pri)
-			if err != nil {
-				return pullRequests, fmt.Errorf("failed to get PR info: %w", err)
-			}
+	//Get pull request IDs
+	pullRequestsOutput, err := a.ccClient.ListPullRequests(&input)
+	if err != nil {
+		return pullRequests, fmt.Errorf("failed to list PRs: %w", err)
+	}
 
-			pullRequests = append(pullRequests, prInfo.PullRequest)
-		}
+	for _, id := range pullRequestsOutput.PullRequestIds {
+		pri := codecommit.GetPullRequestInput{PullRequestId: id}
 
-		if pullRequestsOutput.NextToken == nil {
-			break
+		prInfo, err := a.ccClient.GetPullRequest(&pri)
+		if err != nil {
+			return pullRequests, fmt.Errorf("failed to get PR info: %w", err)
 		}
 
-		input.NextToken = pullRequestsOutput.NextToken
+		pullRequests = append(pullRequests, prInfo.PullRequest)
 	}
 
 	return pullRequests, nil
 }
 
-// sendPushEvent sends an event containing data about a git commit that was
-// pushed to a branch.
-func (a *adapter) sendPushEvent(commit *codecommit.Commit) error {
-	a.logger.Info("Sending Push event")
-
-	event := cloudevents.NewEvent(cloudevents.VersionV1)
-	event.SetType(v1alpha1.AWSEventType(a.arn.Service, pushEventType))
-	event.SetSubject(fmt.Sprintf("%s/%s", a.arn.Resource, a.branch))
-	event.SetSource(a.arn.String())
-	event.SetID(*commit.CommitId)
-	if err := event.SetData(cloudevents.ApplicationJSON, commit); err != nil {
-		return fmt.Errorf("failed to set event data: %w", err)
-	}
-
-	if result := a.ceClient.Send(context.Background(), event); !cloudevents.IsACK(result) {
-		return result
-	}
-	return nil
-}
-
-func (a *adapter) sendPREvent(pullRequest *codecommit.PullRequest) error {
-	a.logger.Info("Sending Pull Request event")
+// sendEvent sends an event containing data about a git commit or PR
+func (a *adapter) sendEvent(codeCommitEvent interface{}) error {
+	a.logger.Info("Sending CodeCommit event")
 
 	event := cloudevents.NewEvent(cloudevents.VersionV1)
-	event.SetType(v1alpha1.AWSEventType(a.arn.Service, prEventType))
 	event.SetSubject(a.branch)
 	event.SetSource(a.arn.String())
-	event.SetID(*pullRequest.PullRequestId)
-	if err := event.SetData(cloudevents.ApplicationJSON, pullRequest); err != nil {
-		return fmt.Errorf("failed to set event data: %w", err)
+
+	switch v := codeCommitEvent.(type) {
+	case *codecommit.PullRequest:
+		event.SetType(v1alpha1.AWSEventType(a.arn.Service, prEventType))
+		err := event.SetData(cloudevents.ApplicationJSON, v)
+		if err != nil {
+			return fmt.Errorf("failed to set event data: %w", err)
+		}
```

Comment:
Does it matter that the interface is asserted here? Can't we move that outside of the switch? `SetData` accepts interfaces.